src/formatting/items.rs

```diff
@@ -2988,13 +2989,36 @@ fn format_header(
     item_name: &str,
     ident: symbol::Ident,
     vis: &ast::Visibility,
+    offset: Indent,
 ) -> String {
-    format!(
-        "{}{}{}",
-        format_visibility(context, vis),
-        item_name,
-        rewrite_ident(context, ident)
-    )
+    let mut result = String::with_capacity(128);
+    let shape = Shape::indented(offset, context.config);
+
+    result.push_str(&format_visibility(context, vis));
+
+    // Check for a missing comment between the visibility and the item name.
+    let after_vis = vis.span.hi();
+    if let Some(before_item_name) = context
+        .snippet_provider
+        .opt_span_before(mk_sp(vis.span().lo(), ident.span.hi()), item_name.trim())
+    {
+        if let Some(cmt) =
+            rewrite_missing_comment(mk_sp(after_vis, before_item_name), shape, context)
+        {
+            result.push_str(&cmt);
+            let need_newline = last_line_contains_single_line_comment(&result);
+            if need_newline {
+                result.push_str(&offset.to_string_with_newline(context.config));
+            } else if cmt.len() > 0 {
+                result.push(' ');
+            }
+        }
+    }
+
+    result.push_str(item_name);
+    result.push_str(&rewrite_ident(context, ident));
+
+    result
```

Comment:
Did you see if using `combine_strs_with_missing_comments` would be viable? Haven't had a chance to look at this too closely yet, but  most cases of comments getting dropped between strings combined with `format!` can be resolved by using `combine_strs_with_missing_comments`

Reply:
Nice tip! Updated to use `combine_strs_with_missing_comments`.